### PR TITLE
[Merged by Bors] - feat(RingTheory/Localization): some API on localizations

### PR DESCRIPTION
**Pull request by chrisflav**
This PR adds some API for localizations of rings. In particular:

- localizations commute with localizations
- localizing away from a product of elements is the same as successive localization at the elements

---

This is used for showing stability properties of the `Locally P` constructor for a property of ring homomorphisms `P`.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — COMMENTED** [review]
LGTM

maintainer merge

File: Mathlib/RingTheory/Localization/Away/Basic.lean [hunk 1]
```diff
@@ -68,6 +69,58 @@ lemma sec_spec (s : S) : s * (algebraMap R S) (x ^ (IsLocalization.Away.sec x s)
...
+    use n, a * u ^ n
+    simp [mul_pow, ← mul_assoc, hn]
+  · obtain ⟨n, hn⟩ := exists_of_eq r hab
+    use n
+    rw [mul_pow, mul_comm (r ^ n), mul_assoc, mul_assoc, hn]
+
+/-- If `r` and `r'` are associated elements of `R`, an `R`-algebra `S`
+is the localization of `R` away from `r` if and only of it is the localization of `R` away from
+`r'`. -/
+lemma iff_of_associated (r r' : R) (h : Associated r r') :
```
Comment (kim-em) [inline Mathlib/RingTheory/Localization/Away/Basic.lean:120]:
```suggestion
lemma iff_of_associated {r r' : R} (h : Associated r r') :
```

**Comment by kim-em** [comment]
bors d+

**Comment by chrisflav** [comment]
Thanks for the reviews!

**Comment by chrisflav** [comment]
Thanks for the reviews!

**Pull request closed without merging**